### PR TITLE
fix(internal/gensupport): Make SendRequestWithRetry check for canceled contexts twice

### PR DESCRIPTION
**Pull request by lhchavez**
This change makes it deterministic to call SendRequestWithRetry with a
canceled context. This is going to be useful because some Cloud APIs
rely on the context being canceled to prevent some operations from
proceeding, like
[`storage.(*ObjectHandle).NewWriter`](https://pkg.go.dev/cloud.google.com/go/storage#ObjectHandle.NewWriter).

Fixes: #1358

### CODE REVIEW

File: internal/gensupport/send_test.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ package gensupport
 
 import (
 	"context"
+	"errors"
```
Comment (codyoss) [inline internal/gensupport/send_test.go:9]: This module still supports 1.11 today. Could you please change this to xerrors.

File: internal/gensupport/send.go [hunk 2]
```diff
@@ -99,6 +99,19 @@ func sendAndRetry(ctx context.Context, client *http.Client, req *http.Request, r
 		case <-time.After(pause):
 		}
 
+		select {
```
Comment (codyoss) [inline internal/gensupport/send.go:102]: Instead of a selecting I think we can just check if ctx.Err() is non-nil.

**Review by codyoss — CHANGES_REQUESTED** [review]
Thanks for the PR and the detailed issue. Just a couple of small things.

File: internal/gensupport/send_test.go (hunk 1 above)
Comment (lhchavez) [reply internal/gensupport/send_test.go]: changed!

File: internal/gensupport/send.go (hunk 2 above)
Comment (lhchavez) [reply internal/gensupport/send.go]: yup, we totally can. TIL that `ctx.Err()` is threadsafe.

**Review by tritone — APPROVED** [review]
Great catch on this. Thanks for the fix!

**Comment by tritone** [comment]
The presubmit is failing but if you run `go mod tidy` that should fix it. @lhchavez

**Comment by lhchavez** [comment]
> The presubmit is failing but if you run `go mod tidy` that should fix it. @lhchavez 

tidied up!

**Review by codyoss — APPROVED** [review]
Thanks, look good to me!

**Pull request merged**